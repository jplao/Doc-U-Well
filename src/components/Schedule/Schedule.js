import React, { Component } from 'react';
import '../../index.scss';
import PropTypes from 'prop-types';
import { addAppointment } from '../../actions/index';
import { addProvider } from '../../actions/index';
import { addInsurance } from '../../actions/index';
import { addProfile } from '../../actions/index';
import { connect } from 'react-redux';

class Schedule extends Component {
  constructor() {
    super()
    this.state = {
      blood: '',
      height: '',
      weight: '',
    }
  }

  render() {
    return(
      <section>
        <form onSubmit={(event) => {
          event.preventDefault()
          this.props.handleProfileSubmit(this.state)}
        }>
          <input placeholder='Blood Type' value={this.state.blood} onChange={(event) => this.setState({blood: event.target.value})}/>
          <input placeholder='Height' value={this.state.height} onChange={(event) => this.setState({height: event.target.value})}/>
          <input placeholder='Weight' value={this.state.weight} onChange={(event) => this.setState({weight: event.target.value})}/>
          <button>Submit</button>
        </form>
      </section>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    appointment: state.appointmentReducer,
    provider: state.providerReducer,
    insurance: state.insuranceReducer,
    profile: state.profileReducer,
    }
}

const mapDispatchToProps = (dispatch) => {
  return {
    // handleAppointmentSubmit: (text => {
    //   dispatch(addAppointment(text))
    // }),
    // handleProviderSubmit: (text => {
    //   dispatch(addProvider(text))
    // }),
    // handleInsuranceSubmit: (text => {
    //   dispatch(addInsurance(text))
    // }),
    handleProfileSubmit: (text => {
      dispatch(addProfile(text))
    }),
  }
}


Schedule.propTypes = {
  type: PropTypes.string
}

export default connect(mapStateToProps, mapDispatchToProps)(Schedule);
