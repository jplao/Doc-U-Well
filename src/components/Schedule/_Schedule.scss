.form-container {
  margin: 20px;
  padding: 10px;
  border: gray solid 2px;
  width: 80%;
  color: #FDF2ea;
  text-size: 1rem;
}

.form-title {
  text-align: center;
  text-size: 1.5rem;
  margin: 10px
}

.input-area {
  margin: 10px;
  padding: 10px;
  line-height: 2rem;
}

.form-btn {
  margin: 10px;
  padding: 10px;
  width: 40%;
  border-radius: 10px;
  color: #301e0a;
  outline: none;
  text-align: center;

  &:hover {
    background-color: #9ec65d;
    color: #fdf2ea;
  }
}