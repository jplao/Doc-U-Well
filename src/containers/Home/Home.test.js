import React from 'react';
import Schedule from './Home';
import { shallow } from 'enzyme';

describe('Home Component', () => {
  it('should match the screenshot', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper).toMatchSnapshot()

  })
})
