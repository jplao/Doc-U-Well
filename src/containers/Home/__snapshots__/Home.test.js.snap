// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container should match the screenshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login"
      >
        <form
          onSubmit={[Function]}
        >
          <h2>
            Sign Up
          </h2>
          Enter e-mail: 
          <input
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
          />
          <br />
          Create password: 
          <input
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
          <br />
          <button>
            Submit
          </button>
        </form>
      </div>,
      "className": "home-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <h2>
            Sign Up
          </h2>
          Enter e-mail: 
          <input
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
          />
          <br />
          Create password: 
          <input
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
          <br />
          <button>
            Submit
          </button>
        </form>,
        "className": "login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Sign Up
            </h2>,
            "Enter e-mail: ",
            <input
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
            />,
            <br />,
            "Create password: ",
            <input
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />,
            <br />,
            <button>
              Submit
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h2",
          },
          "Enter e-mail: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Create password: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login"
        >
          <form
            onSubmit={[Function]}
          >
            <h2>
              Sign Up
            </h2>
            Enter e-mail: 
            <input
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
            />
            <br />
            Create password: 
            <input
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
            <br />
            <button>
              Submit
            </button>
          </form>
        </div>,
        "className": "home-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
          >
            <h2>
              Sign Up
            </h2>
            Enter e-mail: 
            <input
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
            />
            <br />
            Create password: 
            <input
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
            <br />
            <button>
              Submit
            </button>
          </form>,
          "className": "login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Sign Up
              </h2>,
              "Enter e-mail: ",
              <input
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
              />,
              <br />,
              "Create password: ",
              <input
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />,
              <br />,
              <button>
                Submit
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "h2",
            },
            "Enter e-mail: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Create password: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
